packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -69,13 +70,25 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, Tra
     ///         message.
     uint240 internal msgNonce;
 
+    /// @notice Emitted whenever a message is sent to a destination
+    /// @param destination  Chain ID of the destination chain.
+    /// @param target       Target contract or wallet address.
+    /// @param messageNonce Nonce associated with the messsage sent
+    /// @param sender       Address initiating this message call
+    /// @param message      Message payload to call target with.
+    event SentMessage(uint256 indexed destination, address indexed target, uint256 indexed messageNonce, address sender, bytes message);
+
     /// @notice Emitted whenever a message is successfully relayed on this chain.
-    /// @param messageHash Hash of the message that was relayed.
-    event RelayedMessage(bytes32 indexed messageHash);
+    /// @param source       Chain ID of the source chain.
+    /// @param messageNonce Nonce associated with the messsage sent
+    /// @param messageHash  Hash of the message that was relayed.
+    event RelayedMessage(uint256 indexed source, uint256 indexed messageNonce, bytes32 indexed messageHash);
```

Comment:
Do we need anything else besides the nonce here? Given the nonce, you can look up the initiating message to learn all of the other information. Having the hash in the event too i suppose could make sense since its the state change. The source seems like a nice to have in this case

Reply:
Given https://github.com/ethereum-optimism/optimism/pull/11592/files#r1730120603, I think that source, nonce + hash makes sense, I don't think we strictly need the hash

Reply:
agreed the hash isn't strictly necessary since success/failure is delineated by event name. Otherwise the hash is potentially useful querying the `successfulMessage` mapping